docs/infrastructure/deployment-targets/cloud-target-discovery/index.md

```diff
@@ -0,0 +1,98 @@
+
+1. Navigate to **{{Infrastructure,Environments}}**.
+1. Click the ... overflow menu for the environment you want to enable or disable dynamic infrastructure on and select **Edit**.
+1. Expand the **Dynamic infrastructure** section and tick the check-box to enable dynamic infrastructure.
+1. Click **SAVE**.
+
+### Enable deployments without a target
+
+Using target discovery during a deployment means that there may be no existing targets at the start of a deployment. To allow deployments to start without any targets:
+
+1. Navigate to **{{Projects,{Project name},Deployments,Settings}}**.
+1. Expand the **Deployment Targets Required** and select the "Allow deployments to be created when there are no deployment targets" option.
+1. Click **SAVE**.
+
+## Examples
+
+### Azure Web App
+
+Let's say you have an project in Octopus called _Pet Shop_ that deploys an application to an Azure Web App in a _Development_ environment using a role of _PetShopFrontEnd_ and this web app is dynamically created as part of the deployment using an ARM template.
+
+To use this web app previously in Octopus you might have either registered the target manually, or used a [script step](/docs/infrastructure/deployment-targets/dynamic-infrastructure/azure-web-app-target.md) with custom code to try and find and create the web app target. In addition, previously when this web app was no longer needed you might have needed to either [run a script](/docs/infrastructure/deployment-targets/dynamic-infrastructure/remove-octopustarget.md) or manually remove the target in Octopus.
+
+By configuring a well-known variable and tagging your Azure Web App appropriately, Octopus can discover this target for you at deployment time. Additionally, Octopus will continue to monitor the target, and will remove it if it is removed in Azure.
+
+- Configure an [Azure account](/docs/projects/variables/azure-account-variables.md) variable in your project named **Octopus.Azure.Account**, selecting an account that has permissions to be able to find the web app.
+- Add tags to the web app resource within the ARM template to allow Octopus to discover it. For our example we can add the following tags to ensure that it is discovered correctly by our (and only by our project) using [variable substitution](/docs/projects/variables/variable-substitutions.md):
+
+```json
+"resources": [{
+    "type": "Microsoft.Web/sites",
+    //...otherProperties,
+    "tags": {
+        "octopus-environment": "#{Octopus.Environment.Name}",
+        "octopus-role": "PetShopFrontEnd",
+        "octopus-project": "#{Octopus.Project.Name}"
+    }
+}]
+```
+
+Octopus will now discover the web app as a target before deploying to it, matching the environment, role and project from the deployment to the tags created with the ARM template, without any custom scripts or manual registration! Octopus will also remove this target if it is later removed from Azure.
```

Comment:
This might not be an issue, but the last sentence contradicts the hint at the start of the page referring to functionality on removing targets not being available in the EAP. If the functionality will be available before the doco goes live, then no issue. If not, we might want to highlight this again, or change the wording?

```suggestion
Octopus will now discover the web app as a target before deploying to it, matching the environment, role, and project from the deployment to the tags created with the ARM template, without any custom scripts or manual registration! Octopus will also remove this target if it is later removed from Azure in future releases of this feature.
```